Xcode/generate_edition_header.sh

```diff
@@ -0,0 +1,24 @@
+#! /bin/bash -e
+#
+# Xcode build script to generate CBL_Edition.h
+
+# Input and Output file:
+INPUT_FILE="$1"
+OUTPUT_FILE="$2"
+
+# Generated headers directory:
+GEN_HEADERS_DIR="$SRCROOT/Xcode/generated_headers"
+mkdir -p "$GEN_HEADERS_DIR"
+
+# Generated header file:
+GEN_HEADER_FILE="$GEN_HEADERS_DIR/CBL_Edition.h"
+
+# Generate:
+if [ "$CONFIGURATION" == "Debug_EE" ] || [ "$CONFIGURATION" == "Release_EE" ]; then
+  cat "$INPUT_FILE" | sed 's/#cmakedefine/#define/g' > "$GEN_HEADER_FILE"
+else
+  cat "$INPUT_FILE" | sed 's/#cmakedefine\(.*\)/\/\* #undef\1 \*\//g' > "$GEN_HEADER_FILE"
```

Comment:
This won't be correct in the end because you will have undef wrapped in ifndef

Reply:
I emulated the same result of `#cmakedefine COUCHBASE_ENTERPRISE` statement which will be `/* #undef COUCHBASE_ENTERPRISE */` in comment. So basically nothing will be defined or undefined for the case of CE.